packages/components/src/components/SquiggleEditor.tsx

```diff
@@ -29,113 +22,41 @@ const WrappedCodeEditor: React.FC<{
-  diagramStop?: number;
-  /** If the result is a function, how many points along the function it samples */
-  diagramCount?: number;
-  /** When the environment changes. Used again for notebook magic */
-  onChange?(expr: squiggleExpression | undefined): void;
-  /** Previous variable declarations */
-  bindings?: bindings;
-  /** If the output requires monte carlo sampling, the amount of samples */
-  environment?: environment;
-  /** JS Imports */
-  jsImports?: jsImports;
-  /** Whether to show detail about types of the returns, default false */
-  showTypes?: boolean;
-  /** Whether to give users access to graph controls */
-  showControls?: boolean;
-  /** Whether to show a summary table */
-  showSummary?: boolean;
-  /** Whether to log the x coordinate on distribution charts */
-  logX?: boolean;
-  /** Whether to exp the y coordinate on distribution charts */
-  expY?: boolean;
-}
-
-export const SquiggleEditor: React.FC<SquiggleEditorProps> = ({
-  initialSquiggleString = "",
-  width,
-  diagramStart = 0,
-  diagramStop = 10,
-  diagramCount = 20,
-  onChange,
-  bindings = defaultBindings,
-  environment,
-  jsImports = defaultImports,
-  showTypes = false,
-  showControls = false,
-  showSummary = false,
-  logX = false,
-  expY = false,
-}: SquiggleEditorProps) => {
-  const [code, setCode] = useState(initialSquiggleString);
-  React.useEffect(
-    () => setCode(initialSquiggleString),
-    [initialSquiggleString]
-  );
-
-  const { result, observableRef } = useSquiggle({
-    code,
-    bindings,
-    environment,
-    jsImports,
-    onChange,
-  });
-
-  const chartSettings = {
-    start: diagramStart,
-    stop: diagramStop,
-    count: diagramCount,
-  };
+export type SquiggleEditorProps = SquiggleChartProps & {
+  defaultCode?: string;
+  onCodeChange?: (code: string) => void;
+};
 
-  const distributionPlotSettings = {
-    showControls,
-    showSummary,
-    logX,
-    expY,
-  };
+export const SquiggleEditor: React.FC<SquiggleEditorProps> = (props) => {
+  let defaultCode = props.defaultCode ?? "";
+  const [uncontrolledCode, setCode] = useState(defaultCode);
+  let code = props.code ?? uncontrolledCode;
 
+  let chartProps = { ...props, code };
   return (
-    <div ref={observableRef}>
-      <SquiggleContainer>
-        <WrappedCodeEditor code={code} setCode={setCode} />
-        {result.tag === "Ok" ? (
-          <SquiggleItem
-            expression={result.value}
-            width={width}
-            height={200}
-            distributionPlotSettings={distributionPlotSettings}
-            showTypes={showTypes}
-            chartSettings={chartSettings}
-            environment={environment ?? defaultEnvironment}
-          />
-        ) : (
-          <SquiggleErrorAlert error={result.value} />
-        )}
-      </SquiggleContainer>
-    </div>
+    <SquiggleContainer>
+      <WrappedCodeEditor
+        code={code}
+        setCode={(code) => {
+          if (props.onCodeChange) props.onCodeChange(code);
```

Comment:
This can be shortened to `props.onCodeChange?.(code)`, btw.

Reply:
Thank you!!! I was looking for that shorthand